packages/grid/src/DataBarCellRenderer.ts

```diff
@@ -0,0 +1,564 @@
+/* eslint-disable class-methods-use-this */
+import { getOrThrow } from '@deephaven/utils';
+import CellRenderer from './CellRenderer';
+import { isExpandableGridModel } from './ExpandableGridModel';
+import { isDataBarGridModel } from './DataBarGridModel';
+import { VisibleIndex } from './GridMetrics';
+import GridColorUtils, { Oklab } from './GridColorUtils';
+import GridUtils from './GridUtils';
+import memoizeClear from './memoizeClear';
+import { GridRenderState } from './GridRendererTypes';
+
+interface DataBarRenderMetrics {
+  /** The total width the entire bar from the min to max value can take up (rightmostPosition - leftmostPosition) */
+  maxWidth: number;
+  /** The x coordinate of the bar (the left) */
+  x: number;
+  /** The y coordinate of the bar (the top) */
+  y: number;
+  /** The position of the zero line */
+  zeroPosition: number;
+  /** The position of the leftmost point */
+  leftmostPosition: number;
+  /** The position of the rightmost point */
+  rightmostPosition: number;
+  /** The range of values (e.g. max of 100 and min of -50 means range of 150) */
+  totalValueRange: number;
+  /** The width of the databar */
+  dataBarWidth: number;
+  /** The x coordinates of the markers (the left) */
+  markerXs: number[];
+}
+class DataBarCellRenderer extends CellRenderer {
+  drawCellContent(
+    context: CanvasRenderingContext2D,
+    state: GridRenderState,
+    column: VisibleIndex,
+    row: VisibleIndex
+  ) {
+    const { metrics, model, theme } = state;
+    if (!isDataBarGridModel(model)) {
+      return;
+    }
+    const {
+      modelColumns,
+      modelRows,
+      allRowHeights,
+      allRowYs,
+      firstColumn,
+    } = metrics;
+
+    const isFirstColumn = column === firstColumn;
+    const rowHeight = getOrThrow(allRowHeights, row);
+    const modelRow = getOrThrow(modelRows, row);
+    const modelColumn = getOrThrow(modelColumns, column);
+    const rowY = getOrThrow(allRowYs, row);
+    const textAlign = model.textAlignForCell(modelColumn, modelRow);
+    const text = model.textForCell(modelColumn, modelRow);
+    const value = Number(text);
```

Comment:
We should be using `model.valueForCell` here, or even getting the value from `dataBarCellOptions`.
It's possible we have numbers that javascript can't handle so well, and we definitely do not want the renderer assuming what the exact value is from parsing the text.

Reply:
I thought `valueForCell` is only a method on `IrisGrid` models.

Reply:
Okay in that case you should return this value as a number already as part of the dataBarOptions returned by the model.